### DISCUSSION TITLE
send ros2 msg use Foxglove Websocket protocol

### DISCUSSION THREAD

**Discussion by darduck18** [foxglove/community#66] [Q&A] [answered]
Foxglove Websocket protocol support protobuf and json, it is easily bring non-ROS data into Foxglove Studio.
If i want to send ros2 data at the same time,I need to convert ros2 msg to protobuf format,this workload is very heavy to convert all ros2 message. just like(sensor_msgs::msg::PointCloud2 -> ros::sensor_msgs::PointCloud2)
What should I do to send ros2 data use websocket protocol easily? 
or 
Should i change other method?

**Comment by amacneil** [comment, marked as answer]
Please take a look at the [rosbridge](https://foxglove.dev/docs/studio/connection/rosbridge) connection option which works with ROS 2 and does not require any conversion.

**Reply by darduck18** [reply]
thanks for your reply.
i think rosbridge can deal with my issue, Is it feasible to use rosbridge protocol in ws-protocol project?
I want to send both ros2 data and non-ros data at the same websocket server when i select foxglove studio 'websocket' connection.
I according to ws-protocol example to send non-ros data through protobuf format,and i don't know how to send message use "cdr"/"ros2" encoding type.

**Reply by amacneil** [reply]
@jtbandes do we support ros1/ros2/cdr data in Foxglove ws-protocol? would it be hard to add?

**Reply by jtbandes** [reply]
Shouldn't be too hard, we would add them here in this mapping from websocket-style encoding to mcap-style messageEncoding+schemaEncoding: https://github.com/foxglove/studio/blob/a839887b8bcadae6aa77a2e49949e25a837dc310/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L148-L156
